SCClassLibrary/Common/GUI/PlusGUI/Core/ClassBrowser.sc

```diff
@@ -263,7 +263,27 @@ ClassBrowser {
 	searchClasses { |string, rootNumber, matchCase, addHistory = true|
 		var	pool, rootdir, result,
 			searchType = searchMenuKeys[rootNumber] ? \classSearch,
-			isClassSearch;
+			isClassSearch,
+			warning = {
+				var w;
+				w = Window("Invalid search", Rect.aboutPoint(Window.screenBounds.center, 170, 60))
+				.layout_(VLayout(
+					StaticText().align_(\center)
+					.string_("This search requires a class to be selected"),
+					HLayout(
+						StaticText(),  // padding to push button to horiz. center
+						Button().states_([["OK"]])
+						.action_({ w.close })
+						.maxWidth_(80),
+						StaticText()
+					)
+				))
+				.endFrontAction_({
+					w.endFrontAction_(nil).close;
+				})
+				.front;
+				^this  // early exit
```

Comment:
I don't trust using ^ in a function.   it doesn't read clearly and I honestly fear that there are language bugs. 

```^warning.value``` would be better style